### ISSUE TITLE
App crashing when trying to use instantiate and use second instance of Player in iOS 10 and iOS 11.

### ISSUE THREAD

**Issue by nishu-priya** [kaltura/playkit-ios#390]
Steps:
1. Play a video
2. Play another video using another instance of Player
3. The app crashes

Basically, when trying to create multiple players by destroying older players, the app is crashing.
This issue was not happening till v3.14.0

iOS version: **10 and 11(till 11.2)**
Playkit version:  **v3.16.0**

I am getting the below crash log:
```
2020-01-02 08:21:56.162 [Debug] [AVPlayerEngine.swift:276] play() > Play player

2020-01-02 08:21:56.163 [Debug] [AVPlayerEngine.swift:244] deinit > AVPlayerEngine, was deinitialized

2020-01-02 13:51:56.170 CompanyName[25724:171183] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'An instance 0x60000024b820 of class PlayKit.PKAsset was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x600000432040> (

<NSKeyValueObservance 0x600000445520: Observer: 0x60000044b9a0, Key path: status, Options: <New: NO, Old: NO, Prior: NO> Context: 0x0, Property: 0x60000044c840>

)'

*** First throw call stack:

(

0   CoreFoundation                      0x000000011ca0eb0b __exceptionPreprocess + 171

1   libobjc.A.dylib                     0x000000011bd04141 objc_exception_throw + 48

2   CoreFoundation                      0x000000011ca77625 +[NSException raise:format:] + 197

3   Foundation                          0x0000000111eecb53 NSKVODeallocate + 294

4   libobjc.A.dylib                     0x000000011bd18b8e _ZN11objc_object17sidetable_releaseEb + 202

5   PlayKit                             0x00000001141820a1 $s7PlayKit7PKAssetCSgWOh + 17

6   libobjc.A.dylib                     0x000000011bd029bc _ZL27object_cxxDestructFromClassP11objc_objectP10objc_class + 127

7   libobjc.A.dylib                     0x000000011bd0ed34 objc_destructInstance + 129

8   libobjc.A.dylib                     0x000000011bd0ed66 object_dispose + 22

9   AVFoundation                        0x000000011c221a2f -[AVPlayer dealloc] + 1171

10  PlayKit                             0x0000000114195d79 $s7PlayKit14AVPlayerEngineCfD + 313

11  PlayKit                             0x0000000114195f6b $s7PlayKit14AVPlayerEngineCfDTo + 43

12  Foundation                          0x0000000111eeca8d NSKVODeallocate + 96

13  libobjc.A.dylib                     0x000000011bd18b8e _ZN11objc_object17sidetable_releaseEb + 202

14  libobjc.A.dylib                     0x000000011bd192fa _ZN12_GLOBAL__N_119AutoreleasePoolPage3popEPv + 866

15  CoreFoundation                      0x000000011c95de96 _CFAutoreleasePoolPop + 22

16  CoreFoundation                      0x000000011c999aec __CFRunLoopRun + 2172

17  CoreFoundation                      0x000000011c999016 CFRunLoopRunSpecific + 406

18  GraphicsServices                    0x000000011fc95a24 GSEventRunModal + 62

19  UIKit                               0x0000000117260134 UIApplicationMain + 159

20  ComapanyName                           0x000000010baff092 main + 306

21  libdyld.dylib                       0x000000011df9f65d start + 1

)

libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by Manbendra** [comment]
I am also getting this crash with the latest playkit release.

**Comment by x-NR-x** [comment]
Hi @nishu-priya and @Manbendra,

Thank you for contacting us.
A fix has been done, you can find it in [v3.16.1](https://github.com/kaltura/playkit-ios/releases/tag/v3.16.1)

Thanks,
Nilit

**Comment by nishu-priya** [comment]
Thanks for fixing it :)
